File: app/src/main/java/vds/developer/aceworkout/fragments/SummaryFragment.java
package vds.developer.aceworkout.fragments;

import android.graphics.Color;
import android.graphics.Typeface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.design.widget.TabLayout;
import android.support.v4.app.Fragment;
import android.support.v4.view.ViewPager;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import com.github.mikephil.charting.animation.Easing;
import com.github.mikephil.charting.charts.RadarChart;
import com.github.mikephil.charting.components.XAxis;
import com.github.mikephil.charting.components.YAxis;
import com.github.mikephil.charting.data.RadarData;
import com.github.mikephil.charting.data.RadarDataSet;
import com.github.mikephil.charting.data.RadarEntry;
import com.github.mikephil.charting.formatter.ValueFormatter;

import java.util.ArrayList;
import java.util.List;

import vds.developer.aceworkout.Adapters.BottomNavPagerAdapter;
import vds.developer.aceworkout.R;

public class SummaryFragment extends Fragment {

    private RadarChart radarChart;
    private TextView numberDaysTrained;
    private BottomNavPagerAdapter pagerAdapter;
    private ViewPager viewPager;

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater,
                             ViewGroup container,
                             Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.summary_fragment,  container, false);
        radarChart = (RadarChart) view.findViewById(R.id.chart_summaryFragment_radarChart);
        numberDaysTrained = (TextView) view.findViewById(R.id.total_days_trained_number);
        numberDaysTrained.bringToFront();
        setUpRadarChart();
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
    }

    private void setUpRadarChart(){

        int backgroundColor = getResources().getColor(R.color.darkBackgroundColor);
        int textColor = getResources().getColor(R.color.darkTextColor);

        //configure radar chart
        radarChart.setBackgroundColor(backgroundColor);
        radarChart.getDescription().setEnabled(false);
        radarChart.setRotationEnabled(false);

        //color schemes
        radarChart.setWebLineWidth(1F);
        radarChart.setWebColor(textColor);
        radarChart.setWebColorInner(textColor);
        radarChart.setWebAlpha(100);
        radarChart.getLegend().setEnabled(false);


        //animation
        radarChart.animateXY(1400, 1400, Easing.EaseInOutQuad, Easing.EaseInOutQuad);

        //define x axis
        XAxis xAxis = radarChart.getXAxis();
        xAxis.setTextSize(12F);
        xAxis.setYOffset(0);
        xAxis.setXOffset(0);
        xAxis.setValueFormatter(new ValueFormatter(){
            private String[] labels = new String[] {"Back", "Chest", "Arms", "Legs", "Shoulder"};
            @Override
            public String getFormattedValue(float value) {
                return labels[(int) value % labels.length];
            }
        });
        xAxis.setTextColor(textColor);

        YAxis yAxis = radarChart.getYAxis();
        yAxis.setTypeface(Typeface.DEFAULT);
        yAxis.setLabelCount(5, false);
        yAxis.setTextSize(9f);
        yAxis.setAxisMinimum(0f);
        yAxis.setAxisMaximum(80f);
        yAxis.setDrawLabels(false);


        List<RadarEntry> radarSets = generateData(5);

        RadarDataSet set1 = new RadarDataSet(radarSets, "");
        set1.setDrawValues(true);
        set1.setColor(textColor);
        set1.setFillColor(Color.YELLOW);
        set1.setDrawFilled(true);
        set1.setFillAlpha(180);
        set1.setLineWidth(2f);
        set1.setDrawHighlightCircleEnabled(true);
        set1.setDrawHighlightIndicators(false);

        RadarData data = new RadarData(set1);
        data.setValueTypeface(Typeface.DEFAULT);
        data.setValueTextSize(8f);
        data.setDrawValues(false);
        data.setValueTextColor(textColor);
        radarChart.setData(data);
        radarChart.invalidate();


    }

    private List<RadarEntry> generateData(int numLabels){
        float max = 100;
        List<RadarEntry> data = new ArrayList<RadarEntry>();
        for (int i = 0; i<numLabels; i++) {
            data.add( new RadarEntry((float) (Math.random() * max)));
        }
        return data;
    }
}

File: app/src/main/java/vds/developer/aceworkout/fragments/SummaryFragment.kt
package vds.developer.aceworkout.fragments

import android.graphics.Color
import android.graphics.Typeface
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.TextView
import androidx.fragment.app.Fragment
import androidx.viewpager.widget.ViewPager
import com.github.mikephil.charting.animation.Easing
import com.github.mikephil.charting.charts.RadarChart
import com.github.mikephil.charting.data.RadarData
import com.github.mikephil.charting.data.RadarDataSet
import com.github.mikephil.charting.data.RadarEntry
import com.github.mikephil.charting.formatter.ValueFormatter
import vds.developer.aceworkout.Adapters.BottomNavPagerAdapter
import vds.developer.aceworkout.R
import java.util.*

class SummaryFragment : Fragment() {

    private var radarChart: RadarChart? = null
    private var numberDaysTrained: TextView? = null
    private val pagerAdapter: BottomNavPagerAdapter? = null
    private val viewPager: ViewPager? = null

    override fun onCreateView(inflater: LayoutInflater,
                              container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.fragment_summary, container, false)
        radarChart = view.findViewById<View>(R.id.chart_summaryFragment_radarChart) as RadarChart
        numberDaysTrained = view.findViewById<View>(R.id.total_days_trained_number) as TextView
        numberDaysTrained!!.bringToFront()
        setUpRadarChart()
        return view
    }

    private fun setUpRadarChart() {

        val backgroundColor = resources.getColor(R.color.darkBackgroundColor)
        val textColor = resources.getColor(R.color.darkTextColor)

        //configure radar chart
        radarChart!!.setBackgroundColor(backgroundColor)
        radarChart!!.description.isEnabled = false
        radarChart!!.isRotationEnabled = false

        //color schemes
        radarChart!!.webLineWidth = 1f
        radarChart!!.webColor = textColor
        radarChart!!.webColorInner = textColor
        radarChart!!.webAlpha = 100
        radarChart!!.legend.isEnabled = false


        //animation
        radarChart!!.animateXY(1400, 1400, Easing.EaseInOutQuad, Easing.EaseInOutQuad)

        //define x axis
        val xAxis = radarChart!!.xAxis
        xAxis.textSize = 12f
        xAxis.yOffset = 0f
        xAxis.xOffset = 0f
        xAxis.valueFormatter = object : ValueFormatter() {
            private val labels = arrayOf("Back", "Chest", "Arms", "Legs", "Shoulder")
            override fun getFormattedValue(value: Float): String {
                return labels[value.toInt() % labels.size]
            }
        }
        xAxis.textColor = textColor

        val yAxis = radarChart!!.yAxis
        yAxis.typeface = Typeface.DEFAULT
        yAxis.setLabelCount(5, false)
        yAxis.textSize = 9f
        yAxis.axisMinimum = 0f
        yAxis.axisMaximum = 80f
        yAxis.setDrawLabels(false)


        val radarSets = generateData(5)

        val set1 = RadarDataSet(radarSets, "")
        set1.setDrawValues(true)
        set1.color = textColor
        set1.fillColor = Color.YELLOW
        set1.setDrawFilled(true)
        set1.fillAlpha = 180
        set1.lineWidth = 2f
        set1.isDrawHighlightCircleEnabled = true
        set1.setDrawHighlightIndicators(false)

        val data = RadarData(set1)
        data.setValueTypeface(Typeface.DEFAULT)
        data.setValueTextSize(8f)
        data.setDrawValues(false)
        data.setValueTextColor(textColor)
        radarChart!!.data = data
        radarChart!!.invalidate()


    }

    private fun generateData(numLabels: Int): List<RadarEntry> {
        val max = 100f
        val data = ArrayList<RadarEntry>()
        for (i in 0 until numLabels) {
            data.add(RadarEntry((Math.random() * max).toFloat()))
        }
        return data
    }
}
